packages/snaps-controllers/src/interface/utils.ts

```diff
@@ -0,0 +1,93 @@
+import { NodeType, assert } from '@metamask/snaps-sdk';
+import type {
+  Component,
+  Input,
+  FormState,
+  InterfaceState,
+} from '@metamask/snaps-sdk';
+
+/**
+ * Finds the previous value of the input in the old state or sets the input to null.
+ *
+ * @param state - The interface state.
+ * @param component - The Input component.
+ * @returns The input state.
+ */
+export const constructInputState = (
+  state: InterfaceState,
+  component: Input,
+) => {
+  return component.value ?? state[component.name] ?? null;
+};
+
+/**
+ * Finds the previous value of the input in the old state form or sets the input to null.
+ *
+ * @param state - The interface state.
+ * @param component - The Input component.
+ * @param form - The parent form name of the input.
+ * @returns The input state.
+ */
+export const constructFormState = (
```

Comment:
```suggestion
export const constructFormInputState = (
```
Perhaps this explains it better?